src/parse-nm.c

```diff
@@ -552,7 +552,11 @@ parse_tunnels(GKeyFile* kf, NetplanNetDefinition* nd)
                 }
 
                 /* Handle endpoint */
-                wireguard_peer->endpoint = g_key_file_get_string(kf, group, "endpoint", NULL);
+                gchar* endpoint = g_key_file_get_string(kf, group, "endpoint", NULL);
+                if (g_strcmp0(endpoint, "")) {
```

Comment:
This does not catch the `endpoint = NULL` case IMO. g_strcmp0 will return -1 when comparing NULL and the empty string, which resolves to `TRUE`.
```suggestion
                if (endpoint && g_strcmp0(endpoint, "")) {
```

Reply:
Ops, I guess I misinterpreted the man page :P